apps/web/src/hooks/__tests__/useInstrumentVisualization.test.ts

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+
+import { act } from '@testing-library/react';
+import { afterEach, describe, expect, it, vi } from 'vitest';
+
+import { useInstrumentVisualization } from '../useInstrumentVisualization';
+
+const mockUseInstrument = vi.hoisted(() =>
+  vi.fn(() => ({
+    internal: {
+      edition: 1,
+      name: 'test'
+    }
+  }))
+);
+
+const mockRecords = [
+  {
+    __date__: new Date(),
+    __time__: new Date().getTime(),
+    someValue: 'abc'
+  }
+];
+
+vi.mock('@/hooks/useInstrument', () => ({
+  useInstrument: mockUseInstrument
+}));
+
+const mockStore = {
+  currentGroup: { id: 'testGroupId' },
+  currentUser: { username: 'testUser' }
+};
+
+const mockBasicIsoString = '2025-04-30';
+
+const mockUseDownload = vi.fn();
+
+const mockInfoQuery = {
+  useInstrumentInfoQuery: vi.fn()
+};
+
+const mockInstrumentRecords = {
+  useInstrumentRecords: vi.fn()
+};
+
+vi.mock('@/store', () => ({
+  useAppStore: vi.fn((selector) => selector(mockStore))
+}));
+
+vi.mock('@douglasneuroinformatics/libui/hooks', () => ({
+  useDownload: () => mockUseDownload,
+  useNotificationsStore: () => ({ addNotification: vi.fn() }),
+  useTranslation: () => ({ t: vi.fn((key) => key) })
+}));
+
+vi.mock('react', async (importOriginal) => {
+  const actual = await importOriginal<typeof React>();
+  return {
+    ...actual,
+    useEffect: vi.fn(),
+    useMemo: vi.fn(),
+    useState: vi.fn(() => [mockRecords, 'setRecords'])
+  };
+});
+
+vi.mock('@/hooks/useInstrumentInfoQuery', () => ({
+  useInstrumentInfoQuery: () => mockInfoQuery
+}));
+
+vi.mock('@/hooks/useInstrumentRecords', () => ({
+  useInstrumentRecords: () => mockInstrumentRecords
+}));
+
+vi.mock('@douglasneuroinformatics/libjs', () => ({
+  toBasicISOString: vi.fn(() => mockBasicIsoString)
+}));
+
+describe('useInstrumentVisualization tests', () => {
```

Comment:
remove "tests", we always just use describe('{NAME OF FUNCTION}'). Otherwise looks good.